packages/wasm-bindings/src/lib.rs

```diff
@@ -7,6 +7,82 @@ use wasm_bindgen::prelude::*;
 #[global_allocator]
 static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
 
+/// Modify the currently selected tool in the document state store
+#[wasm_bindgen]
+pub fn select_tool(tool_id: u32) {
+	todo!()
+}
+
+/// Modify the currently selected Document in the editor state store
+// TODO: should it be possible to open multiple documents at once?
+#[wasm_bindgen]
+pub fn select_document(document_id: u32) {
```

Comment:
The Editor Library should be document-agnostic. You might have two different document viewports open side-by-side. I believe the Event Dispatcher can infer which document the event is referring to based on the mouse hover status (`panel_enter_hover(...)`). Then the Event Dispatcher sends either an Action or Operation. Operations all include the Document ID. And Actions that pertain to the Open Documents State Store also include the Document ID.